### ISSUE TITLE
Passing "pyproject.toml" when calling black using the API

### ISSUE THREAD

**Issue by khrynczenko** [psf/black#3177]
Hello,
I am trying to use `black` as a library and so far so good. It is easy to just call `black.format_str(code_block, mode=Mode())` but I would like to pass a configuration as an argument as well. Is there any way I can create `Mode` object from a `pyproject.toml` file? Preferably I would like not to parse the file myself and create `Mode` object manually. So far I could find anything related to such thing inside the source code.

TLDR: I would like to call `black.format_str` in a way that it uses a `pyproject.toml` configuration specified by myself.

**Comment by JelleZijlstra** [comment]
We don't currently have a documented API; #779 is tracking that. Closing as you've solved your immediate problem and #779 tracks the bigger enhancement request.
